builder/simplify.cc

```diff
@@ -409,40 +473,40 @@ class simplifier : public node_mutator {
     void learn_from_equal(const expr& a, const expr& b) {
       if (const variable* v = a.as<variable>()) {
         // bounds of a are [b, b].
-        k.push_back(set_value_in_scope(bounds, v->sym, point(b)));
+        k.push_back(set_value_in_scope(info_map, v->sym, {point(b), modulus_remainder()}));
```

Comment:
I'm wondering, in expressions like this, what if `info_map` already has some useful alignment info in it? Isn't this dropping that?

Rather than a ton of boilerplate to preserve it, I wonder if we should not put both of these in one struct, and instead have `symbol_map<interval_expr> bounds; symbol_map<modulus_remainder> alignment;` ?

Maybe this can't happen due to the lack of shadowing? (BTW, our banning of shadowing is not total yet, so maybe an assert that there is no shadowing would fail, but would still be OK in theory).

Reply:
I guess we can carry over alignment from the current value, it still should be correct, even though the bounds have changed. What do you think?

Reply:
Okay, I added that.